crates/refunder/src/refund_service.rs

```diff
@@ -98,63 +93,55 @@ impl RefundService {
     async fn identify_uids_refunding_status_via_web3_calls(
         &self,
         refundable_order_uids: Vec<EthOrderPlacement>,
-    ) -> Result<Vec<OrderUid>> {
+    ) -> Vec<(OrderUid, Vec<CoWSwapEthFlowAddress>)> {
         let mut batch = Web3CallBatch::new(self.web3.transport().clone());
         let futures = refundable_order_uids
             .iter()
             .map(|eth_order_placement| {
                 let order_hash: [u8; 32] = eth_order_placement.uid.0[0..32]
                     .try_into()
                     .expect("order_uid slice with incorrect length");
-                let order = self
-                    .ethflow_contract
-                    .orders(ethcontract::tokens::Bytes(order_hash))
-                    .batch_call(&mut batch);
+
+                let contract_calls = self
+                    .ethflow_contracts
+                    .iter()
+                    .map(|contract| {
+                        contract
+                            .orders(ethcontract::tokens::Bytes(order_hash))
+                            .batch_call(&mut batch)
```

Comment:
The `refunder` uses our auto batching logic under the hood. If make use of this this function would probably end up being a lot easier to read.

Reply:
Not sure what are you referring to exactly.

Reply:
The underlying `HttpTransport` automatically batches individual RPC calls together. So instead of manually calling `.batch_call()` you can simple do things like `join_all(rpc_calls)` and it will do the same thing while being a lot easier to read.